bundles/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/handler/GroheOndusBaseHandler.java

```diff
@@ -142,14 +152,23 @@ protected Location getLocation() {
     protected @Nullable T getAppliance(OndusService ondusService) {
         try {
             BaseAppliance appliance = ondusService.getAppliance(getRoom(), config.applianceId).orElse(null);
-            if (appliance.getType() != getType()) {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                        "Thing is not a GROHE SENSE Guard device.");
-                return null;
+            if (appliance != null) {
+                if (appliance.getType() != getType()) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Thing is not a GROHE SENSE Guard device.");
+                    return null;
+                }
+                return (T) appliance;
+            } else {
+                logger.debug("getAppliance for thing {} returned null", thing.getUID());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Error retrieving appliance data from service");
```

Comment:
Please move text to src/main/resources/OH-INF/i18n/groheondus.properties to allow for localization.